File: src/api/activity.ts
import apiClient from './client';

export interface AuditLogApiData {
  id: number;
  user_id?: number;
  username?: string;
  client_id?: number;
  client_identifier?: string;
  action: string;
  resource_type: string;
  resource_id?: string;
  ip_address?: string;
  details?: string;
  created_at: string;
  severity: string;
}

export const activityApi = {
  list: async (params?: {
    client?: string;
    severity?: string;
    action?: string;
    date_from?: string;
    date_to?: string;
    limit?: number;
  }) => {
    const res = await apiClient.get('/activity', { params });
    return res.data;
  }
};

File: server/app/schemas/activity.py
import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict

class AuditLogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: Optional[int] = None
    username: Optional[str] = None
    client_id: Optional[int] = None
    client_identifier: Optional[str] = None
    action: str
    resource_type: str
    resource_id: Optional[str] = None
    ip_address: Optional[str] = None
    details: Optional[str] = None
    created_at: datetime.datetime
    severity: str = "INFO"  # derived or passed: INFO, WARNING, ERROR
